### PR TITLE
[RDY] More fixes to the new TUI

### LINKED ISSUE #2005: [bug] Background color redraw issue with term module removal and termkey/unibilium changes
I am confirming the issue reported by @bigeagle at his last comment in #1820 (and referencing #1997 because of @tarruda's continued work):

> Hi, using the merged version, there're still a problem that the terminal's background color cannot be automatically "resetted". To repeat this bug, nvim -u NONE and try to show some error message, e.g. :q on an unsaved file and there would be an error message with red bg color, now press : and the red background is still there.
> I'm using xterm-256color.
> 687474703a2f2f692e696d6775722e636f6d2f3358514a44735a2e706e67

I am noticing this bug also affects redrawing past the color column
```viml
if exists('+colorcolumn')
    set colorcolumn=81
else
    ...
```

And I see the following in my terminal:
screen shot 2015-02-17 at 2 45 10 pm

The following are details of my Neovim version and platform:

    NVIM 0.0.0-alpha+201502170110 (compiled Feb 17 2015 01:51:39)
    Commit: 66a03a7385dfd11503d22d8cc9dc6a972e27dcc2
    Build type: RelWithDebInfo
    Compilation: /usr/local/Library/ENV/4.3/clang -Wconversion -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/tmp/neovim-ijeEOa/build/config -I/tmp/neovim-ijeEOa/src -I/tmp/neovim-ijeEOa/.deps/usr/include -I/tmp/neovim-ijeEOa/.deps/usr/include -I/tmp/neovim-ijeEOa/.deps/usr/include/luajit-2.0 -I/tmp/neovim-ijeEOa/.deps/usr/include -I/usr/local/opt/gettext/include -I/usr/include -I/tmp/neovim-ijeEOa/build/src/nvim/auto -I/tmp/neovim-ijeEOa/build/include

    Installed With: Homebrew (using neovim/homebrew-neovim tap from the neovim repository)
    Platform: OS X 10.9.5
    uname -a: Darwin 13.4.0 Darwin Kernel Version 13.4.0: Wed Dec 17 19:05:52 PST 2014; root:xnu-2422.115.10~1/RELEASE_X86_64 x86_64
    Shell: zsh running in iTerm2

@bigeagle seems to be running into this error with a build of nvim on a Linux variant.

### LINKED ISSUE #2024: Copying and pasting from system clipboard to command line adds trailing `<F37>`
Basically, I select some text to automatically copy it to the system clipboard and the paste it to the command line using <kbd>cmd</kbd><kbd>v</kbd> I get a trailing `<F37>` string/character.

This with `nvim -u NONE` using https://github.com/neovim/neovim/commit/b1d079c83b3dd459114d1ba7c2ff1b07a9ee3e9e and iTerm2 (Build 2.0.0.20141103).

screen_shot_2015-02-19_at_12_01_46

### PR DESCRIPTION
**Pull request by tarruda**
Fix #2005 #2016 #2017 #2024

### CODE REVIEW

**Comment by renstrom** [comment]
Verified.

File: src/nvim/tui/term_input.inl [hunk 1]
```diff
@@ -259,8 +265,8 @@ static TermInput *term_input_new(void)
   int curflags = termkey_get_canonflags(rv->tk);
   termkey_set_canonflags(rv->tk, curflags | TERMKEY_CANON_DELBS);
   // setup input handle
-  uv_tty_init(uv_default_loop(), &rv->input_handle, rv->in_fd, 1);
-  uv_tty_set_mode(&rv->input_handle, UV_TTY_MODE_RAW);
+  uv_pipe_init(uv_default_loop(), &rv->input_handle, 0);
+  uv_pipe_open(&rv->input_handle, 0);
```
Comment (oni-link) [inline src/nvim/tui/term_input.inl]: Is `in_fd` not used anymore?

Comment (tarruda) [inline src/nvim/tui/term_input.inl]: forgot about it :+1:

**Comment by tarruda** [comment]
merging after tests

**Pull request merged**